### ISSUE TITLE
False Positive

### ISSUE THREAD

**Issue by machineghost** [twolfson/css-validator#11]
I came across a pretty strange error using this validator.  The following code:

    
    p {
        line-height:24px;
    }

(note the blank line before the start of the CSS) generates this result:

    { validity: false,
      errors:
       [ { line: '0',
           errortype: 'parse-error',
           context: '\n                            ',
           errorsubtype:
            '\n                                skippedString\n                            ',
           skippedstring:
            '\n                                Lexical error at line 1, column 1.  Encountered: "" (0), after : ""\n                            ',
           type: 'generator.unrecognize',
           message:
            '\n        \n                                Parse Error\n                            ',
           error: '\n                        ' } ],
      warnings: [] }

However, running that exact code through the official W3C web validator (https://jigsaw.w3.org/css-validator/validator) returns no errors.  And removing the empty line before the code removes the error.

I shouldn't have to say this but ... it's allowed in CSS to have blank lines before your styles start!

**Comment by twolfson** [comment]
Hmm, that is strange and I'm seeing the same issue on the website. I'm really curious what it could be since we're more/less an HTTP request wrapper. Digging into this further

**Comment by twolfson** [comment]
I'm unable to reproduce the issue locally. Here's me forcing an error:

```bash
$ cat tmp.css 

p {{
    line-height:24px;
}

$ node bin/css-validator tmp.css 
tmp.css:4: 
    Parse Error
$
```

and here's me with the submitted error case:

```bash
$ cat tmp.css 

p {
    line-height:24px;
}

$ node bin/css-validator tmp.css 
$
```

I even tried a bunch of space/empty line combinations as well as an empty file but have been unable to get it to trigger

Can you provide me the following info:

- What version of Node.js are you running?
- What OS are you running?
- How are you running `css-validator`? Is it in another project? On the CLI? Programatically?

**Comment by machineghost** [comment]
- What version of Node.js are you running?

v11.7.0

   - What OS are you running?

Linux (doubt this matters)

   - How are you running css-validator? Is it in another project? On the
   CLI? Programatically?

Programatically.  I read in a file and then fed the contents to the
validator.  I couldn't figure out why the file was getting errors, so I
deleted down to the simplest working part, and then undid the last change
(the blank line before the selector) and saw it give me the error again.

I can sympathize if you aren't able to repro from that, but unfortunately
that's all I can offer.

**Comment by twolfson** [comment]
OS matters in case there's different line endings (e.g. `\r\n` for Windows)

Cool, I'm running Node.js@v8.12.0

I'll poke around some more with this new info. Thanks!

**Comment by machineghost** [comment]
If it helps this also generates an error:

    body, html {
        margin:0;
        height:100%;
        background-color:#E8E8E4;
        font-size:14px;
    }
    p {
        line-height:24px;
    }

With the same error:

    [ { line: '0',
        errortype: 'parse-error',
        context: '\n                            ',
        errorsubtype:
         '\n                                skippedString\n                            ',
        skippedstring:
         '\n                                Lexical error at line 1, column 1.  Encountered: "" (0), after : ""\n                            ',
        type: 'generator.unrecognize',
        message:
         '\n        \n                                Parse Error\n                            ',
        error: '\n                        ' } ]

**Comment by twolfson** [comment]
Yep, tried CLI and programmatic usage in 11.7.0. No issues still:

```js
$ cat tmp.js 
var validateCss = require('./');

validateCss({
  text: `

body, html {
    margin:0;
    height:100%;
    background-color:#E8E8E4;
    font-size:14px;
}
p {
    line-height:24px;
}`
}, function (err, data) {
  console.log(err, data);
});
$ n use 11.7.0 tmp.js 
null { validity: true, errors: [], warnings: [] }
```

How are you running it programmatically?

**Comment by machineghost** [comment]
My code is literally just:

const validateCss = require('css-validator');

const allCss = await getCombinedCss(cssFileNames);
const cssValidationErrors = await getCssValidationErrors(allCss);

(followed by a console.log of the errors) where "getCombinedCss" is just a
function that combines all of a student's CSS files into a single string.

**Comment by twolfson** [comment]
Sorry, I can't do much else without seeing the rest of the code or finding out that the issue is something else (e.g. generators are misbehaving, file contents are loading async, file contents are a buffer instead of a string)

I'd recommend copy/pasting the code snippets I've provided and verifying there are no issues with those. If there isn't, then it's likely something besides the newline (maybe GitHub is removing hidden characters on paste)

Going to close the issue for now =/
